libgnucash/engine/engine-interface.scm

```diff
@@ -136,7 +136,7 @@
   (make-record-type
    "gnc:transaction-structure"
    '(transaction-guid currency date-entered date-posted
-                      num description notes split-scms)))
+                      num description notes association split-scms)))
```

Comment:
I'm not sure I agree on the reasoning in your commit message. I do understand you logic and follow it to some extent.
However to my mind cut == copy + delete and duplicate == copy + paste. By only making cut transfer the associated file you're breaking this universal semantic. Which may cause more confusion.

Your reasoning also assumes the user wants to associate unique files to transactions. But that may not be the case. Like I can imagine the user wants to attach a pdf of a bank statement to all transactions associated with that particular bank statement.

I think this is something too user dependent for gnucash to decide and I would propose the paste transaction (and by extension the duplicate transaction code) checks if a file association is present and if so ask the user what to do with it: add it to the new transaction as well or not.

Reply:
After some more thinking, I believe that if you are cutting/copying and then pasting a transaction you should expect all information for that transaction so I would propose that be the case with the associated entry. With the duplicate action, it already brings up a dialogue for cheque number and date so I think we can add to that, if there is an associated entry, add a tick box with 'Keep Associated Entry' with the default being ticked.

Reply:
I have just noticed one other scenario, auto complete, should the associated entry be copied then, I am thinking not.

Reply:
No, autocomplete shoud not copy the associated entry.